app/models/shipit/stack.rb

```diff
@@ -58,6 +58,7 @@ def lock_author=(user)
     validates :repo_owner, format: {with: /\A[a-z0-9_\-\.]+\z/}, length: {maximum: REPO_OWNER_MAX_SIZE}
     validates :repo_name, format: {with: /\A[a-z0-9_\-\.]+\z/}, length: {maximum: REPO_NAME_MAX_SIZE}
     validates :environment, format: {with: /\A[a-z0-9\-_\:]+\z/}, length: {maximum: ENVIRONMENT_MAX_SIZE}
+    validates :deploy_url, format: {with: URI.regexp(%w(http https))}, allow_blank: true
```

Comment:
We have a couple stacks without a web frontend, so we set an `ssh://` url. I think we should allow it.